app/jobs/add_workflow_job.rb

```diff
@@ -2,26 +2,29 @@
 
 ##
 # Job to add a workflow to and object
-class AddWorkflowJob < GenericJob
-  attr_reader :workflow_name
+class AddWorkflowJob < BulkActionJob
+  def workflow_name
+    params.fetch(:workflow)
```

Comment:
A bulk action might have parameters, which are made available to BulkActionItems (see delegate below).